cmd/statusd/main.go

```diff
@@ -158,6 +168,28 @@ func startCollectingStats(interruptCh <-chan struct{}, nodeManager common.NodeMa
 	}
 }
 
+func syncAndStopNode(nodeManager common.NodeManager) (err error) {
+	log.Println("Node will synchronize and exit")
+	v := *syncAndExit
+	if v < 0 {
+		return errors.New("negative timeout value")
+	}
+	if v == 0 {
+		err = nodeManager.EnsureSync(context.Background())
+	} else {
+		ctx, cancel := context.WithTimeout(context.Background(), (time.Duration)(v)*time.Minute)
+		err = nodeManager.EnsureSync(ctx)
+		defer cancel()
+	}
+	if err != nil {
```

Comment:
I think you can skip this error check. It's ok to properly stop a node even if it didn't manage to sync the chain.

Reply:
Alright, then I definitely need to use `(github.com/pkg/errors).Wrap` here to return combined errors. Shall I `dep ensure --add` that?

Reply:
isn't `errors.Wrap` a standard library function?

Reply:
Looks like it isn't. I also got confused for a moment: https://golang.org/pkg/errors/

Reply:
@adambabik @mandrigin Would you mind adding this as a dependency?

Reply:
Yeah, I think it makes sense to add it. It is useful in many places. @adambabik what do you think?

Reply:
An alternative is moving stopping node to `main.go`. Wrapping both errors together is also not really correct because the sync error has nothing to do with stop node error.

Maybe something like this:
```go
func ensureSync(nodeManager common.NodeManager, timeout int) error {
  // it's not possible to pass negative timeout from the condition in main.go

  if timeout == 0 {
    return nodeManager.EnsureSync(context.Background())
  }

  ctx, cancel := context.WithTimeout(context.Background(), (time.Duration)(timeout)*time.Minute)
  defer cancel()

  return nodeManager.EnsureSync(ctx)
}

// in main.go
if *syncAndExit >= 0 {
  var exitCode int

  if err := ensureSync(backend.NodeManager(), *syncAndExit); err != nil {
    log.Printf("Failed to sync the chain: %v", err)
    exitCode = 1
  }

  done, err := backend.NodeManager().StopNode()
  if err != nil {
    log.Fatalf("Failed to stop node: %v", err)
  }
  <-done

  os.Exit(exitCode)
}
```

Ugh, I am not happy with this neither :(